tensorflow/python/kernel_tests/shape_ops_test.py

```diff
@@ -82,6 +113,14 @@ def testBasic(self):
     self._testAll(np.zeros([2, 3, 5, 7, 11]))
     self._testAll(np.zeros([2, 3, 5, 7, 11, 13]))
 
+  def testBasicSparse(self):
```

Comment:
I'd suggest removing this (and also see the above comment about `_testCpuSparse()`).  The code path should look like:

`testBasic() -> _testAll() -> _testCpu()  -> _compareRankSparse()`.